lib/messages.d.ts

```diff
@@ -0,0 +1,13 @@
+//
+// automatically generated code; do not edit manually!
+//
+///<reference path=".d.ts"/>
+interface IMessages{
```

Comment:
Unfortunately I was not able to reuse any parts of the interface due to the following scenario:
These interfaces are generated based upon json files - one default one in the common lib and client-specific ones inside each client respectively. This means that a following case is possible:
Common json file:
```JavaScript
{ 
    MessagesObject: {
        Message1: "Some message",
        Message2: "Some other message"
    }
}
```

Client json file:
```JavaScript
{ 
    MessagesObject: {
        Message1: "Some overriden message"
    }
}
```

This would mean that the interface, generated from the client json file won't be able to extend the interface, generated from the common json because it wouldn't compile.

Personally I don't feel like this is a major issue, because all of this code is generated anyways.